### ISSUE TITLE
[BUG] Get "https://registry.redhat.io/v2/": dial tcp: lookup registry.redhat.io on 10.217.4.10:53: server misbehaving

### ISSUE THREAD

**Issue by sandrodesantis** [code-ready/crc#2175]
### General information
  * OS: macOS 
  * Hypervisor: hyperkit
  * Did you run `crc setup` before starting it (Yes/No)? yes
  * Running CRC on: Laptop 

## CRC version
```bash
# Put `crc version` output here
CodeReady Containers version: 1.24.0+5f06e84b
OpenShift version: 4.7.2 (embedded in executable)
  
## CRC status
```bash
# Put `crc status --log-level debug` output here
CRC VM:          Running
OpenShift:       Running (v4.7.2)
Disk Usage:      11.81GB of 32.74GB (Inside the CRC VM)
Cache Usage:     13.36GB
Cache Directory: /Users/sandrodesantis/.crc/cache

## CRC config
```bash
# Put `crc config view` output here
- consent-telemetry                     : yes

## Host Operating System
```bash
# Put the output of `cat /etc/os-release` in case of Linux
# put the output of `sw_vers` in case of Mac
# Put the output of `systeminfo` in case of Windows
ProductName:	macOS
ProductVersion:	11.2.3
BuildVersion:	20D91

### Steps to reproduce
  1.  Download local crc, via https://cloud.redhat.com/openshift/create/local
  2. Extract files
  3. cp crc /usr/local/bin
  4. crc setup
  5. crc start -p pull-secret.txt
  6. eval $(crc oc-env)
  7. oc login -u developer https://api.crc.testing:6443
  8. oc new-project pj1
  9. oc new-app php name=php1   => creating some application
10. oc get all => check pods
11. oc describe pod/php-84688b96b9-whvr4 => check pod with issue

### Expected
The download of images does not work, looks like https://registry.redhat.io/v2/ is not reached

### Actual

oc get co
NAME                                       VERSION   AVAILABLE   PROGRESSING   DEGRADED   SINCE
authentication                             4.7.2     True        False         False      31s
config-operator                            4.7.2     True        False         False      13d
console                                    4.7.2     True        False         False      6m9s
dns                                        4.7.2     True        False         False      7m6s
etcd                                       4.7.2     True        False         False      13d
image-registry                             4.7.2     True        False         False      6m23s
ingress                                    4.7.2     True        False         False      12d
kube-apiserver                             4.7.2     True        False         False      13d
kube-controller-manager                    4.7.2     True        False         False      13d
kube-scheduler                             4.7.2     True        False         False      13d
machine-approver                           4.7.2     True        False         False      13d
marketplace                                4.7.2     True        False         False      6m43s
network                                    4.7.2     True        False         False      13d
node-tuning                                4.7.2     True        False         False      7m38s
openshift-apiserver                        4.7.2     True        False         False      6m10s
openshift-controller-manager               4.7.2     True        False         False      6m28s
openshift-samples                          4.7.2     True        False         False      13d
operator-lifecycle-manager                 4.7.2     True        False         False      13d
operator-lifecycle-manager-catalog         4.7.2     True        False         False      13d
operator-lifecycle-manager-packageserver   4.7.2     True        False         False      7m2s
service-ca                                 4.7.2     True        False         False      13d


NAME                       READY   STATUS             RESTARTS   AGE
pod/php-84688b96b9-whvr4   0/1     ImagePullBackOff   0          34m

ssh core@api.crc.testing
core@api.crc.testing: Permission denied (publickey,gssapi-keyex,gssapi-with-mic)

ssh core@api.crc.testing
core@api.crc.testing: Permission denied (publickey,gssapi-keyex,gssapi-with-mic).


### Logs
Events:
  Type     Reason          Age                    From               Message
  ----     ------          ----                   ----               -------
  Normal   Scheduled       28m                    default-scheduler  Successfully assigned pj1/php-84688b96b9-whvr4 to crc-rsppg-master-0
  Normal   AddedInterface  28m                    multus             Add eth0 [10.217.0.52/23]
  Normal   Pulling         26m (x4 over 28m)      kubelet            Pulling image "image-registry.openshift-image-registry.svc:5000/openshift/php@sha256:856653890008f036ac1eb36751f524be44fc03942bfa0b9ef000fa3560139d0a"
  Warning  Failed          26m (x4 over 28m)      kubelet            Failed to pull image "image-registry.openshift-image-registry.svc:5000/openshift/php@sha256:856653890008f036ac1eb36751f524be44fc03942bfa0b9ef000fa3560139d0a": rpc error: code = Unknown desc = Error reading manifest sha256:856653890008f036ac1eb36751f524be44fc03942bfa0b9ef000fa3560139d0a in image-registry.openshift-image-registry.svc:5000/openshift/php: unknown: **unable to pull manifest from registry.redhat.io/ubi8/php-74@sha256:856653890008f036ac1eb36751f524be44fc03942bfa0b9ef000fa3560139d0a: Get "https://registry.redhat.io/v2/": dial tcp: lookup registry.redhat.io on 10.217.4.10:53: server misbehaving**
  Warning  Failed          26m (x4 over 28m)      kubelet            Error: **ErrImagePull**
  Warning  Failed          8m21s (x85 over 28m)   kubelet            Error: **ImagePullBackOff**
  Normal   BackOff         3m26s (x106 over 28m)  kubelet            Back-off pulling image "image-registry.openshift-image-registry.svc:5000/openshift/php@sha256:856653890008f036ac1eb36751f524be44fc03942bfa0b9ef000fa3560139d0a"

Before gather the logs try following if that fix your issue
```bash
$ crc delete -f
$ crc cleanup
$ crc setup
$ crc start --log-level debug
```

Please consider posting the output of `crc start --log-level debug`  on http://gist.github.com/ and post the link in the issue.
[crc.log](https://github.com/code-ready/crc/files/6238969/crc.log)

**Comment by praveenkumar** [comment]
@sandrodesantis this looks like dns issue can you do following and let us know if that works for you?

```
$ crc delete
$ crc start -n 1.1.1.1
```

**Comment by sandrodesantis** [comment]
Hi @praveenkumar , tks .. using this parameter worked, but, why is necessary this parameter?

**Comment by sandrodesantis** [comment]
Another question, Only works without VPN active, there is another configuration to work with VPN? tks.

**Comment by praveenkumar** [comment]
@sandrodesantis Please try https://github.com/code-ready/crc/wiki/VPN-support--with-an--userland-network-stack one for vpn side and provide us the feedback.

**Comment by DanielViglione** [comment]
I can confirm that 

```
crc start -n 1.1.1.1
```
resolved that error, although now I am getting:

```
error: build error: error building at STEP "RUN /usr/libexec/s2i/assemble": error while running runtime: exit status 18
```

**Comment by cfergeau** [comment]
@DanielViglione I'd recommend opening a new issue for this, with more details about what you did before getting this error.
